manage_app_pool - add periodic recycle control

lcoulson opened the pull request:
manage_app_pool - add periodic recycle control

new param left default undef
logic in class handles undef

caller set to 0 to disable periodic app pool recycling

minutes value converted to ticks and given to powershell for timespan
constructor as int64 max value is powerhsell $([int64]::MaxValue)

validation will reject negatives or out of max bound

juniorsysadmin commented on manifests/manage_app_pool.pp:
```diff
@@ -16,15 +15,23 @@
   validate_bool($rapid_fail_protection)
 
+  if $apppoolrecycleperiodicminutes != undef {
+    validate_integer($apppoolrecycleperiodicminutes, 15372286728, 0) # powershell $([int64]::MaxValue) / 600000000, we're not dealing with negative
```
You probably should make these numbers constants, variables or parameters, defined here or elsewhere

juniorsysadmin commented on manifests/manage_app_pool.pp:
```diff
@@ -16,15 +15,23 @@
 
+  if $apppoolrecycleperiodicminutes != undef {
+    validate_integer($apppoolrecycleperiodicminutes, 15372286728, 0) # powershell $([int64]::MaxValue) / 600000000, we're not dealing with negative
+    $periodicticks        = $apppoolrecycleperiodicminutes * 600000000
```
As above

lcoulson commented:
@igalic @juniorsysadmin - added tests, standardized variable naming, added documentation example, updated stdlib min version to allow for validate_integer()

juniorsysadmin commented:
Looks okay to me
